### ISSUE TITLE
Creating more than one instance of Parser causes a segfault

### ISSUE THREAD

**Issue by Anahkiasen** [php-mime-mail-parser/php-mime-mail-parser#84]
Works:
```php
$parser = new Parser;
$parser->setPath('mail.txt');
$parser->setPath('mail.txt');
```

Segault:
```php
(new Parser())->setPath('mail.txt');
(new Parser())->setPath('mail.txt');
```

```
$ php --version
PHP 7.0.5 (cli) (built: Mar 31 2016 20:27:43) ( ZTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies
    with Xdebug v2.4.0, Copyright (c) 2002-2016, by Derick Rethans
```

**Comment by Anahkiasen** [comment]
The mail I'm trying to parse:

[1-before.txt](https://github.com/php-mime-mail-parser/php-mime-mail-parser/files/201374/1-before.txt)

**Comment by eXorus** [comment]
no issue for me :

        $Parser1 = (new Parser())->setPath($file);
        $Parser2 = (new Parser())->setPath($file);
PHP 5.6.11-1ubuntu3.1 (cli) 
Copyright (c) 1997-2015 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2015 Zend Technologies
    with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2015, by Zend Technologies

I think it's an issue with mailparse 3.0 on PHP7. Could you go try this :

        $mailparseObject = mailparse_msg_parse_file($file);
        $mailparseObjectParsed = mailparse_msg_get_structure($mailparseObject);
        $mailparseObject = mailparse_msg_parse_file($file);
        $mailparseObjectParsed = mailparse_msg_get_structure($mailparseObject);

If one of this failed I think we need to open an issue on https://github.com/php/pecl-mail-mailparse

**Comment by Anahkiasen** [comment]
Hm no the above example works fine, it's only when using the class itself:

```php
<?php
use PhpMimeMailParser\Parser;

require 'vendor/autoload.php';

$file = __DIR__.'/tests/_pages/messages/4-after.txt';

// Works
$mailparseObject = mailparse_msg_parse_file($file);
$mailparseObjectParsed = mailparse_msg_get_structure($mailparseObject);
$mailparseObject = mailparse_msg_parse_file($file);
$mailparseObjectParsed = mailparse_msg_get_structure($mailparseObject);

// Segfault
(new Parser())->setPath($file)->getMessageBody();
(new Parser())->setPath($file)->getMessageBody();
```

**Comment by eXorus** [comment]
Could you check :

$mailparseObject = mailparse_msg_parse_file($file);
$mailparseObjectParsed = mailparse_msg_get_structure($mailparseObject);
$arrayResult = array();
foreach ($mailparseObjectParsed as $part_id) {
   $part = mailparse_msg_get_part($mailparseObject, $part_id);
   $arrayResult[$part_id] = mailparse_msg_get_part_data($part);
}

$mailparseObject = mailparse_msg_parse_file($file);
$mailparseObjectParsed = mailparse_msg_get_structure($mailparseObject);
$arrayResult = array();
foreach ($mailparseObjectParsed as $part_id) {
   $part = mailparse_msg_get_part($mailparseObject, $part_id);
   $arrayResult[$part_id] = mailparse_msg_get_part_data($part);
}

**Comment by Anahkiasen** [comment]
Both seem to work, even dumped `$arrayResult` to check if it wasn't failing silently but I do get the correct results. 

My setup isn't anything particular apart from maybe the ZTS, it's a standard `brew install php70 --with-zts php70-mailparse`

**Comment by eXorus** [comment]
Sorry but I can't reproduce

Like you can see all tests are OK (I added your test CreatingMoreThanOneInstanceOfParser)
https://travis-ci.org/php-mime-mail-parser/php-mime-mail-parser/builds/122433346

**Comment by Anahkiasen** [comment]
Well I solved this another way on my end anyway (bound the parser as a singleton in my container). But I'm guessing it's because the PHP versions on Travis aren't ZTS maybe

**Comment by mta59066** [comment]
I realize this is a closed issue and all tests are passing on Travis, but wanted add additional info in case something else comes up in the future.

I have the same problem, doesn't have to be a second call, even the first call to mailparse_msg_parse_file() within Parser->setPath() causes segfault. No problem if mailparse_msg_parse_file() is called on its own outside Parser class.

Running this on OSX as the OP installed via brew, but unlike OP not ZTS.

```
$ php --version
PHP 7.0.6 (cli) (built: May 14 2016 18:44:52) ( NTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies
```

**Comment by mta59066** [comment]
Problem is due to mailparse_msg_free() as the following segfaults as well. 

```
$res = mailparse_msg_parse_file($file);
mailparse_msg_free($res);
```

Looks like it was fixed on 3.0.1 
https://pecl.php.net/package-changelog.php?package=mailparse&release=3.0.1
